biblib/views/library_view.py

```diff
@@ -277,156 +696,58 @@ def get(self, library):
+
+        # Get library id
+        library_id, library_error = self.get_library_from_slug(user, library)
+        if library_error: 
+            return library_error
+    
+        # If set to True, return notes in library
+        notes = request.args.get('notes', type=bool, default=True)
+ 
+        # Get user id for service 
+        service_uid = self.get_service_uid(user) 
+
+        # Data needed to process the library request
+        data = {"user": user, 
+                "service_uid": service_uid, 
+                "library_id": library_id,
+                "start": start, 
+                "rows": rows, 
+                "sort": sort, 
+                "fl": fl, 
+                "raw_library": raw_library,
+                "notes": notes}
+        
+        with current_app.session_scope() as session:
+            data['session'] = session
+            library, response, solr_error = self.process_library_request(data)
+            if solr_error: 
+                return solr_error
 
-        # Skip anymore logic if the library is public or the exception token is present
-        special_token = current_app.config.get('READONLY_ALL_LIBRARIES_TOKEN')
-        if library.public or (special_token and request.headers.get('Authorization', '').endswith(special_token)):
+        # Skip any more logic if the library is public or the exception token is present
+        if self.is_library_public_or_has_special_token(library, request):
             current_app.logger.info('Library: {0} is public'
-                                    .format(library.id))
+                                    .format(library_id))
             return response, 200
-        else:
-            current_app.logger.warning('Library: {0} is private'
-                                       .format(library.id))
-
-        # If the user does not exist then there are no associated permissions
-        # If the user exists, they will have permissions
-        if self.helper_user_exists(absolute_uid=user):
-            service_uid = \
-                self.helper_absolute_uid_to_service_uid(absolute_uid=user)
-        else:
-            current_app.logger.error('User:{0} does not exist in the database.'
-                                     ' Therefore will not have extra '
-                                     'privileges to view the library: {1}'
-                                     .format(user, library.id))
-
-            return err(NO_PERMISSION_ERROR)
-
-        # If they do not have access, exit
-        if not self.read_access(service_uid=service_uid,
-                                library_id=library.id):
-            current_app.logger.error(
-                'User: {0} does not have access to library: {1}. DENIED'
-                .format(service_uid, library.id)
-            )
-            return err(NO_PERMISSION_ERROR)
+        
+        current_app.logger.warning('Library: {0} is private'.format(library_id))
```

Comment:
I know this is pre-existing code, but it's weird that it's generating a warning-level log for this - can you downgrade it to info-level?